xlive/H2MOD/Modules/EventHandler/EventHandler.h

```diff
@@ -0,0 +1,200 @@
+	{
+		this->name = name;
+		this->gameState = gameState;
+		this->callback = callback;
+	}
+};
+struct NetworkPeerEventCallback
+{
+	
+	std::string name;
+	std::function<void(int)> callback;
+	/**
+	 * \brief Basic callback for when a peer action happens in the network session.
+	 * \param name A name that is stored with this callback that can later be referenced for removal.
+	 * \param callback a void that will be executed when this event happens the paramter in is the peer index.
+	 */
+	NetworkPeerEventCallback(std::string name, std::function<void(int)> callback){
+		this->name = name;
+		this->callback = callback;
+	}
+};
+
+namespace EventHandler
+{
+	/**
+	 * \brief Adds a callback to be executed when the GameState changes
+	 * \param callback A GameStateCallback to be executed when the event happens.
+	 * \param threaded Tells the EventHandler whether or not this callback should be ran in its own thread with other threaded callbacks.
+	 */
+	void registerGameStateCallback(GameStateCallback callback, bool threaded);
+	/**
+	 * \brief Removes a callback from the store
+	 * \param name Name of the callback to remove.
+	 */
+	void removeGameStateCallback(std::string name);
+	/**
+	 * \brief Executes the callbacks that match the given GameState.
+	 * \param gamestate
+	 */
+	void executeGameStateCallbacks(BYTE gamestate);
+
+	/**
+	 * \brief Adds a callback to be executed when the NetworkSession recieves a player_add packet
+	 * \param callback A NetworkPeerEventCallback to be executed when the event happens.
+	 * \param threaded Tells the EventHandler whether or not this callback should be ran in its own thread with other threaded callbacks.
+	 */
+	void registerNetworkPlayerAddCallback(NetworkPeerEventCallback callback, bool threaded);
+	/**
+	 * \brief Removes a callback from the store
+	 * \param name Name of the callback to remove
+	 */
+	void removeNetworkPlayerAddCallback(std::string name);
+	/**
+	 * \brief Executes the callbacks passing the Peer Index to the function
+	 * \param peerIndex 
+	 */
+	void executeNetworkPlayerAddCallbacks(int peerIndex);
+
+	/**
+	 * \brief Adds a callback to be executed when the NetworkSession recieves a player_remove packet
+	 * \param callback A NetworkPeerEventCallback to be executed when the event happens.
+	 * \param threaded Tells the EventHandler whether or not this callback should be ran in its own thread with other threaded callbacks.
+	 */
+	void registerNetworkPlayerRemoveCallback(NetworkPeerEventCallback callback, bool threaded);
+	/**
+	 * \brief Removes a callback from the store
+	 * \param name Name of the callback to remove
+	 */
+	void removeNetworkPlayerRemoveCallback(std::string name);
+	/**
+	 * \brief Executes the callbacks passing the Peer Index to the function
+	 * \param peerIndex 
+	 */
+	void executeNetworkPlayerRemoveCallbacks(int peerIndex);
+
+	///////////////////////////////////////////////
+	//Someone smarter than me can figure this out//
```

Comment:
no commented out code, if someone else is supposed to finish this, then this shouldn't be up for PR yet

Reply:
Future me is smarter than me, it was meant for myself. I can remove it I guess.